production/grafanacloud-install.sh

```diff
@@ -73,7 +73,21 @@ main() {
   esac
 
   log '--- Retrieving config and placing in /etc/grafana-agent.yaml'
-  retrieve_config | sudo tee /etc/grafana-agent.yaml
+  # The named pipe "$t/config" is created below so we can tee the stdout of
+  # retrieve_config to /etc/grafana-agent.yaml with sudo privileges.
+  # This is a POSIX-compliant work around for dash not having pipefail shellopt.
+  local t
+  t=$(mktemp -d)
+  mkfifo "$t/config"
+  # SC2024 can be safely ignored as we want to redirect using the current user's
+  # permissions. See https://github.com/koalaman/shellcheck/issues/2070
+  # shellcheck disable=SC2024
+  sudo tee /etc/grafana-agent.yaml < "$t/config"  &
+  if ! retrieve_config 2>/dev/null 1>"$t/config"; then
+    fatal 'Failed to retrieve config'
```

Comment:
Would it make sense if `fatal` was changed to [signal an abort to the top-level script](https://stackoverflow.com/a/9894126)? That might help clean this up a little bit and make it easier to follow, if it fixes the problem at all.

Reply:
@rfratto that would work as well. would you prefer I update this to use that approach?

Reply:
I think it might be easier to follow, but I'll leave it up to you. LMK when you want me to merge 👍

Reply:
updated 👍